src/navier.f90

```diff
@@ -1280,5 +1289,255 @@ subroutine tbl_flrt (ux1,uy1,uz1)
     enddo
 
   end subroutine tbl_flrt
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !!
+  !!  subroutine: pipe_bulk / pipe_bulk_u / pipe_bulk_phi
+  !!      AUTHOR: Rodrigo Vicente Cruz
+  !! DESCRIPTION: Correction of pipe's bulk velocity (constant 
+  !!              flow rate) and bulk temperature
+  !!
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !********************************************************************
+  !
+  subroutine pipe_bulk(ux,uy,uz,ep)
+  !
+  !********************************************************************
+
+    use param
+    use decomp_2d
+    use variables
+    use var,  only: phi1
+
+    implicit none
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3))  :: ux,uy,uz,ep
+    integer                                             :: is
+
+    !Bulk temperature correction
+    if (numscalar.ne.0) then
+        do is=1,numscalar
+            call pipe_bulk_phi(phi1(:,:,:,is),ux,ep,is,one)
+        enddo
+    endif
+
+    !Bulk velocity correction
+    call pipe_bulk_u(ux,uy,uz,ep,one)
+
+  end subroutine pipe_bulk
+  !********************************************************************
+  !
+  subroutine pipe_bulk_u(ux,uy,uz,ep,constant)
+  !
+  !********************************************************************
+
+    use decomp_2d
+    use variables
+    use param
+    use var
+    use ibm_param, only: ra
+    use MPI
+
+    implicit none
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3))  :: ux,uy,uz,ep
+    real(mytype)                                        :: constant !bulk velocity value
```

Comment:
Could you rename `constant` to `bulk_u_constant` or something? Otherwise it is not obvious when used in code what it is.

I would also add the `intent(in)` attribute as the subroutine should not change its value.
Similarly I would separate the `ep` from `ux,uy,uz` as `ep` should not be modified it is also `intent(in)`, the others should be `intent(inout)` as their values get modified.
These help both the reader to understand, and also allows the compiler to enforce coding intentions (and potentially catch errors).

Reply:
Thank you @pbartholomew08, these points have been solved in 6780e3678ad72d0b9f50aa1685c35358692bc854.